removing the docker exp docker file `Dockerfile.exp`

valeriupredoi opened the pull request:
<!--
    Thank you for contributing to our project!

    Please do not delete this text completely, but read the text below and keep
    items that seem relevant. If in doubt, just keep everything and add your
    own text at the top, a reviewer will update the checklist for you.

    While the checklist is intended to be filled in by the technical and scientific
    reviewers, it is the responsibility of the author of the pull request to make
    sure all items on it are properly implemented.
-->

## Description

<!--
    Please describe your changes here, especially focusing on why this pull request makes
    ESMValTool better and what problem it solves.

    Before you start, please read our contribution guidelines: https://docs.esmvaltool.org/en/latest/community/

    Please fill in the GitHub issue that is closed by this pull request, e.g. Closes #1903
-->

@sloosvel mentioned in https://github.com/ESMValGroup/ESMValTool/issues/2532#issuecomment-1039961445 that this is not needed anymore by BSC and as far as I know that was the only use of this docker file, so finally removing it for good :partying_face: 

- Closes #2532 


* * *

## Before you get started

<!--
    Please discuss your idea with the development team before getting started,
    to avoid disappointment or unnecessary work later. The way to do this is
    to open a new issue on GitHub.
-->

- [x] [☝ Create an issue](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#contributing-code-and-documentation) to discuss what you are going to do

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

zklaus reviewed (changes requested):
Wait a minute

valeriupredoi commented:
> Wait a minute

no rush :racehorse:

zklaus commented:
Of course. We'll certainly learn something when we understand why the build fails and removing the container doesn't address any problem we have at the moment, so I would keep it around for now and focus on more pressing matters.

valeriupredoi commented:
sounds good to me! I tried to understand why the build picks up an older `environment.yml` file - that was the root cause of `rasterio` being wheel-built from PyPi, but beats me if I know why...

valeriupredoi commented:
OK I fixed the darn build and added a GA test for its build (so we hit the fan before it actually ships to DockerHub) in https://github.com/ESMValGroup/ESMValTool/pull/2551 so  closing this
